### DISCUSSION TITLE
Custom docker-compose file match not honoured

### DISCUSSION THREAD

**Discussion by mschoettle** [renovatebot/renovate#20080] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

34.115.1

### If you're self-hosting Renovate, select which platform you are using.

Gitea or Forgejo

### If you're self-hosting Renovate, tell us what version of the platform you run.

1.17.3

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I started containerizing my Wordpress page and put the `wp-content` (with Wordpress plugins and theme) into a git repo. Unfortunately, two of the plugins contain a docker compose file (among other things).

To avoid getting dependency updates for those I changed the renovate config to

```json
"enabledManagers": ["dockerfile", "docker-compose"],
  "docker-compose": {
    "fileMatch": ["^docker-compose.yml$"]
  },
```

However, I still receive dependency updates after setting it.

### Relevant debug logs

<details><summary>Logs</summary>

```
{"name":"renovate","hostname":"cd1df91e9cd0","pid":10,"level":20,"logContext":"iVH9gkel2IaF569B8XUUa","repository":"my/repo","msg":"Applying enabledManagers filtering","time":"2023-01-28T02:39:26.114Z","v":0}
{"name":"renovate","hostname":"cd1df91e9cd0","pid":10,"level":20,"logContext":"iVH9gkel2IaF569B8XUUa","repository":"my/repo","msg":"Using file match: (^|/)(?:docker-)?compose[^/]*\\.ya?ml$ for manager docker-compose","time":"2023-01-28T02:39:26.227Z","v":0}
{"name":"renovate","hostname":"cd1df91e9cd0","pid":10,"level":20,"logContext":"iVH9gkel2IaF569B8XUUa","repository":"my/repo","msg":"Using file match: ^docker-compose.yml$ for manager docker-compose","time":"2023-01-28T02:39:26.229Z","v":0}
{"name":"renovate","hostname":"cd1df91e9cd0","pid":10,"level":20,"logContext":"iVH9gkel2IaF569B8XUUa","repository":"my/repo","msg":"Using file match: (^|/|\\.)Dockerfile$ for manager dockerfile","time":"2023-01-28T02:39:26.275Z","v":0}
{"name":"renovate","hostname":"cd1df91e9cd0","pid":10,"level":20,"logContext":"iVH9gkel2IaF569B8XUUa","repository":"my/repo","msg":"Using file match: (^|/)Dockerfile[^/]*$ for manager dockerfile","time":"2023-01-28T02:39:26.276Z","v":0}
{"name":"renovate","hostname":"cd1df91e9cd0","pid":10,"level":20,"logContext":"iVH9gkel2IaF569B8XUUa","repository":"my/repo","msg":"Matched 4 file(s) for manager docker-compose: docker-compose.prod.yml, docker-compose.yml, wp-content/plugins/papercite/docker-compose.phpunit.yml, wp-content/plugins/papercite/docker-compose.yml","time":"2023-01-28T02:39:26.277Z","v":0}
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by rarkins** [comment, marked as answer]
[fileMatch](https://docs.renovatebot.com/configuration-options/#filematch) is a mergeable field, meaning you can _add_ patterns to it, but not remove default ones. Instead, you need to use `ignorePaths` or `includePaths`.

PRs welcome if you'd like to improve the documentation to note this explicitly for future useres

**Reply by mschoettle** [reply]
Thanks!

Does `ignorePaths` need to be defined at the root of the config or can it be defined within a manager?

> PRs welcome if you'd like to improve the documentation to note this explicitly for future useres

I will do that. 

At the same time, what do you think about an additional debug log along the lines of "Using merged file matches for manager docker-compose: (^|/)(?:docker-)?compose[^/]*\.ya?ml$,^docker-compose.yml$"? This would make it clear at that level that the file matches are merged.

**Reply by rarkins** [reply]
`ignorePaths` can be within a manager.

The file match for each manager is already logged. You're welcome to submit a PR to enhance the log to note when it's merged or not.
